plugins/cpp/parser/src/diagnosticmessagehandler.cpp

```diff
@@ -0,0 +1,76 @@
+#include <llvm/ADT/SmallString.h>
+#include <cppparser/filelocutil.h>
+#include "diagnosticmessagehandler.h"
+
+namespace cc
+{
+namespace parser
+{
+
+DiagnosticMessageHandler::DiagnosticMessageHandler(
+  SourceManager& srcMgr_,
+  std::shared_ptr<odb::database> db_)
+    : _srcMgr(srcMgr_), _db(db_)
+{
+}
+
+void DiagnosticMessageHandler::HandleDiagnostic(
+  clang::DiagnosticsEngine::Level diagLevel_,
+  const clang::Diagnostic& info_)
+{
+  // TODO: Is this needed?
+  clang::DiagnosticConsumer::HandleDiagnostic(diagLevel_, info_);
```

Comment:
This only increments the `numWarnings` or `numErrors` counters. We do not use them now, so technically it is not needed, but later it could be useful, so in my opinion we shall keep it and remove the TODO.